inductiva/utils/grids.py

```diff
@@ -0,0 +1,43 @@
+"""Utility functions to manipulate grid data."""
+import typing
+
+import scipy
+import numpy as np
+
+
+def get_meshgrid(x_range: typing.Sequence[float],
+                 y_range: typing.Sequence[float], x_num: int, y_num: int):
+
+    x_grid = np.linspace(*x_range, num=x_num)
+    y_grid = np.linspace(*y_range, num=y_num)
+
+    x_grid, y_grid = np.meshgrid(x_grid, y_grid, indexing="ij")
+
+    return x_grid, y_grid
+
+
+def interpolate_between_grids(x_num: int, y_num: int, z_array: np.ndarray):
+    """Interpolate between two grid with different resolution.
+    
+    Args:
+        x_num: New number of grid points in x-direction.
+        y_num: New number of grid points in y-direction.
+        z_array: Values on the z-axis of the previous grid.
+    """
+
+    x_grid_prev, y_grid_prev = np.meshgrid(np.arange(z_array.shape[0]),
```

Comment:
out of curiosity: do you need to do this? Can you get x_grid_prev and y_grid_prev directly from z_array? (I am just curious to know)

Reply:
The z_array does not carry information about the specific values of the x_grid_prev...
The way to get it is the len of the array.